common/reviews/api/heft.api.md

```diff
@@ -341,14 +340,10 @@ export interface IPreCompileSubstage extends IBuildSubstage<BuildSubstageHooksBa
 }
 
 // @beta
-export interface IRegisterParameterOptions extends IBaseCommandLineDefinition {
-    associatedActionNames: string[];
-}
+export type IRegisterParameterOptions = IBaseCommandLineDefinition & IParameterAssociatedActionNames;
 
 // @beta
-export interface IRegisterParameterWithArgumentOptions extends IBaseCommandLineDefinitionWithArgument {
-    associatedActionNames: string[];
-}
+export type IRegisterParameterWithArgumentOptions = IBaseCommandLineDefinitionWithArgument & IParameterAssociatedActionNames;
 
 // @beta
 export interface IRunScriptOptions<TStageProperties> {
```

Comment:
From the sample project:
```ts
          if (customParameter.actionAssociated && customParameter.valueProvided) {
            const customContent: string = `${customStringParameter.value?.repeat(
              customNumberParameter.value || 1
            )}_${customStringListParameter.value?.join('_')}`;
            await FileSystem.writeFileAsync(`${buildFolder}/lib/custom_output.txt`, customContent, {
              ensureFolderExists: true
            });
          }
```

This code uses the parsed values, but what it's doing doesn't make any sense?  When I built the sample project, it wrote `testtesttesttesttest_eevee_togepi_mareep` to the output file. 😋  

Could we make the example slightly more intuitive, maybe just writing the parameter values on separate lines?

Reply:
Ok. I guess that would be more business-like. Eevee, Togepi, and Mareep made it in the test file because a certain little someone is particular fond of those pokemon—probably not a good reason though. :)